concepts/observability-configurable-logging/07-multiple-logpipeline-investigation.md

```diff
@@ -0,0 +1,106 @@
+
+### Case 2
+![a](./assets/logpipeline-invstigation/case-2/case-2.svg)
+
+Setup:
+1. one input and 2 outputs (loki (fluentbit-loki plugin) + mockserver) with rewrite-tags
+2. logpipelines: [loki](./assets/logpipeline-invstigation/case-2/loki.yaml), [mock-server](./assets/logpipeline-invstigation/case-2/mockserver.yml)
+
+
+Result:
+1. Output chunking keeps with 150M of newest data.
+2.  The tail plugin sends the data to next phase of pipeline (filter plugin). If the outputs is not working then only the buffer at the rewrite tag would be filled and tail plugin buffer wont be filled
+3. When the buffer is full then we see some error logs stating that and the old logs are discarded
+    ```unix
+    [2022/04/21 14:38:23] [error] [input:emitter:log_emitter] error registering chunk with tag: log_rewritten
+    [2022/04/21 14:38:24] [error] [input chunk] chunk 1-1650551903.999375896.flb would exceed total limit size in plugin http.1
+    ```
+4. When grafana-loki plugin is stopped it stopped all the pipelines (so mockserver which was fine was also stopped) 
+5. When mock server was down the logs were still shipped to loki
+
+
+### Case 3
+
+Setup:
+1. one input with 2 outputs (mockserver + mockserver) with rewrite tags
+2. logpipelines: [mockserver-1](./assets/logpipeline-invstigation/case-3/mockserver-1.yml), [mockserver-2](./assets/logpipeline-invstigation/case-3/mockserver-2.yml)
+
+
+Result
+1. When both outputs are down then the buffer was filled and eventually the fluentbit pod got killed because of 500 (most probably because of CPU throttling)
+2. Tail plugin kept pushing logs to rewrite buffer and they were eventually lost
+
+### Case 4
+Setup:
+1. one input with 2 outputs (loki (with official loki plugin) + mockserver) with rewrite tags
+2. logpipelines: [loki](./assets/logpipeline-invstigation/case-4/loki.yml), [mockserver](./assets/logpipeline-invstigation/case-4/mock-server.yml)
+
+
+Result
+1. Mockserver was down and the loki output was still working
+2. The chunks in the filesystem buffer are rolled  (the old chunks deleted new ones created)
+3. When loki output was down the mockserver output was still working
+
+![mockserver-down](/assets/logpipeline-invstigation/case-4/dashboard-mock-down.png)
+
+
+![loki-down](/assets/logpipeline-invstigation/case-4/dashboard-loki-down.png)
+
+### Case 5
+Setup:
+1. one input with 2 outputs (loki (with official loki plugin) + mockserver) without rewrite tags
+2. logpipelines: [loki](./assets/logpipeline-invstigation/case-5/loki.yml), [mockserver](./assets/logpipeline-invstigation/case-5/mock-server.yml)
+Result
+1. Mockserver was down, this led to filling of the tail buffer
+2. When the tail plugin buffer is filled, it kept still reading.
+3. The chunks in tail plugin is rolled (the old chunks deleted new ones created)
+4. Loki output stopped working as well
+5. Found following [issue](https://github.com/fluent/fluent-bit/issues/4373) in github which describes the same problem.
```

Comment:
That's not a result of the test though. It's a result of your research.
I suggest separating it from the results list with a blank line.

```suggestion

A known [Fluent Bit GitHub issue](https://github.com/fluent/fluent-bit/issues/4373) describes the same problem.
```